### ISSUE TITLE
Android envConfigFiles does not find correct envFile if buildTypes or productFlavors contain non alphabets

### ISSUE THREAD

**Issue by sckoh** [luggit/react-native-config#333]
envConfigFiles does not match if buildTypes or productFlavors contain non alphabets

```
project.ext.envConfigFiles = [
    armv7debug: ".env.development",
]

buildTypes {
    beta {
    }
}

productFlavos {
    armv7 {
    }
}
```

the task name will be :AppName:assembleArmv7Beta and the result currentFlavor will be "armv"

it is caused by the regex Pattern in method getCurrentFlavor only match for alphabet

```
def getCurrentFlavor() {
    Gradle gradle = getGradle()

    // match optional modules followed by the task
    // (?:.*:)* is a non-capturing group to skip any :foo:bar: if they exist
    // *[a-z]+([A-Za-z]+) will capture the flavor part of the task name onward (e.g., assembleRelease --> Release)
    def pattern = Pattern.compile("(?:.*:)*[a-z]+([A-Z][A-Za-z]+)")
    def flavor = ""

    gradle.getStartParameter().getTaskNames().any { name ->
        Matcher matcher = pattern.matcher(name)
        if (matcher.find()) {
            flavor = matcher.group(1).toLowerCase()
            return true
        }
    }

    return flavor
}
```

**Comment by michbil** [comment]
Current flavour detecton function is pretty bad actually, it often fails during android studio sync, etc:
I found this one on the stackoverflow, replaced it in dotenv.gradle and it's working much better:
```
def getCurrentFlavor() {
    Gradle gradle = getGradle()
    String  tskReqStr = gradle.getStartParameter().getTaskRequests().toString()

    Pattern pattern;

    if( tskReqStr.contains( "assemble" ) )
        pattern = Pattern.compile("assemble(\\w+)(Release|Debug)")
    else
        pattern = Pattern.compile("generate(\\w+)(Release|Debug)")

    Matcher matcher = pattern.matcher( tskReqStr )

    if( matcher.find() )
        return matcher.group(1).toLowerCase()
    else
    {
        println "NO MATCH FOUND"
        return "";
    }
}
```

**Comment by r0b0t3d** [comment]
I tried add some log
```
gradle.getStartParameter().getTaskNames().any { name ->
   println("Name: $name")
   ...
}
```
Output:
```
Name: clean
Name: :react-native-linear-gradient:generateDebugSources
```

Above script worked for me too.

**Comment by antitoine** [comment]
Thank @sckoh this solves my issue. Your PR need to be merged quickly...
Until this append, I use your fix in PR #334 with this patch package : https://github.com/ds300/patch-package
So I keep the last version of react-native-config with your patch.

**Comment by luancurti** [comment]
Close by PR #334
